### PR TITLE
:recycle: Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by tkoyama010**
:recycle: Hi all. I moved Travis to GitHub Actions. WFM. I tested this yml in [mayavi-doc-translations](https://github.com/getfem-doc/mayavi-doc-translations/blob/master/.github/workflows/main.yml).

### CODE REVIEW

**Comment by tkoyama010** [comment]
If you merge this PR, could you please set the following key at setting tab?
* [ ] GITHUB_TOKEN
* [ ] PERSONAL_ACCESSTOKEN
* [ ] SSH_PRIVATE_KEY
* [ ] TX_TOKEN

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        NAME: "sphinx-auto-update"
+        EMAIL: "sphinx-auto-update"
+        REPO: sphinx-doc-translations
+        JOB_ID: ${{ steps.jobs.outputs.job_id }}
+        HTML_URL: ${{ steps.jobs.outputs.html_url }}
+        SPHINXINTL_TRANSIFEX_USERNAME: sphinxjp
+        SPHINXINTL_TRANSIFEX_PROJECT_NAME: sphinx-doc
+        TX_TOKEN: ${{ secrets.TX_TOKEN }}
+      run: |
+        sudo apt-get install python3-venv
```
Comment (tk0miya) [inline .github/workflows/main.yml:38]: Why did you use venv here?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        source ../.venv/bin/activate
+        git submodule update
+        (cd sphinx; git fetch origin; git checkout master; git reset --hard origin/master; git branch -a)
+        pip install -U pip setuptools
+        pip install -r ./requirements.txt
+        sh ./locale/update.sh
+        git branch -a
+        git checkout master
+        git config --global user.email $EMAIL
+        git config --global user.name $NAME
```
Comment (tk0miya) [inline .github/workflows/main.yml:49]: It seems the "update" action is a bit large and complicated. How abot moving these git settings to independent action?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        git add .
+        git commit --allow-empty -m ":green_heart: $JOB_ID
```
Comment (tk0miya) [inline .github/workflows/main.yml:51]: It seems the original message contains "[ci skip]". Why removing?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        JOB_ID: ${{ steps.jobs.outputs.job_id }}
+        HTML_URL: ${{ steps.jobs.outputs.html_url }}
+        SPHINXINTL_TRANSIFEX_USERNAME: sphinxjp
+        SPHINXINTL_TRANSIFEX_PROJECT_NAME: sphinx-doc
+        TX_TOKEN: ${{ secrets.TX_TOKEN }}
+      run: |
+        sudo apt-get install python3-venv
+        python3 -m venv ../.venv
+        source ../.venv/bin/activate
+        git submodule update
```
Comment (tk0miya) [inline .github/workflows/main.yml:41]: Is this still needed? It seems you're already using `submodules: true` option of actions/checkout action.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    - name: Setup SSH
+      uses: MrSquaare/ssh-setup-action@v1
+      with:
+        host: github.com
+        private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+    - name: update
+      env:
+        ACCOUNT: sphinxjp
+        ORGANIZATION: sphinx-doc
+        NAME: "sphinx-auto-update"
```
Comment (tk0miya) [inline .github/workflows/main.yml:29]: It seems GHAction provides `GITHUB_REPOSITORY` envvar for such purpose.
https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

File: .github/workflows/main.yml (hunk 1 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: You are right :+1: I changed to set path to modules.

File: .github/workflows/main.yml (hunk 2 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: That's a good idea. I split :scissors: it to "commit" action.

File: .github/workflows/main.yml (hunk 3 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: Oops. I forgot to reflect it.

File: .github/workflows/main.yml (hunk 4 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: You are right. I removed it :wastebasket:

File: .github/workflows/main.yml (hunk 5 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: Oh. I didn't know that ! Thanks :bow:

**Comment by tkoyama010** [comment]
@tk0miya Thank you very much for your review :coffee:

File: locale/update.sh [hunk 6]
```diff
@@ -10,12 +10,12 @@ set -ex
 
 # pull po files from transifex
 cd `dirname $0`
-sphinx-intl create-transifexrc
+/home/runner/.local/bin/sphinx-intl create-transifexrc
```
Comment (tk0miya) [inline locale/update.sh:13]: -1 for hard code. It makes harder to run this script on local. Please set `$PATH` instead.

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    - name: update
+      env:
+        ACCOUNT: sphinxjp
+        JOB_ID: ${{ steps.jobs.outputs.job_id }}
+        HTML_URL: ${{ steps.jobs.outputs.html_url }}
+        SPHINXINTL_TRANSIFEX_USERNAME: sphinxjp
+        SPHINXINTL_TRANSIFEX_PROJECT_NAME: sphinx-doc
+        TX_TOKEN: ${{ secrets.TX_TOKEN }}
+      run: |
+        source ../.venv/bin/activate
```
Comment (tk0miya) [inline .github/workflows/main.yml:34]: What are you doing here?

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        token: ${{ secrets.PERSONAL_ACCESSTOKEN }}
+    - name: Setup SSH
+      uses: MrSquaare/ssh-setup-action@v1
+      with:
+        host: github.com
+        private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+    - name: update
+      env:
+        ACCOUNT: sphinxjp
+        JOB_ID: ${{ steps.jobs.outputs.job_id }}
```
Comment (tk0miya) [inline .github/workflows/main.yml:28]: If my understanding correct, some enviroment variables should move to the "commit" action.

Comment (tkoyama010) [reply .github/workflows/main.yml]: You're right.

File: .github/workflows/main.yml (hunk 7 above)
Comment (tkoyama010) [reply .github/workflows/main.yml]: Thanks!

File: locale/update.sh (hunk 6 above)
Comment (tkoyama010) [reply locale/update.sh]: You're right :+1:

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      uses: MrSquaare/ssh-setup-action@v1
+      with:
+        host: github.com
+        private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+    - name: install
+      run: |
+        (cd sphinx; git fetch origin; git checkout master; git reset --hard origin/master; git branch -a)
+        pip3 install -U pip setuptools
+        pip3 install -r ./requirements.txt
+    - name: script
```
Comment (tk0miya) [inline .github/workflows/main.yml:30]: I feel this name is strange. Each step is a script!
How about "update" or "update transifex pot and local po files" (comment in update.sh).

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      env:
+        TX_TOKEN: ${{ secrets.TX_TOKEN }}
+        SPHINXINTL_TRANSIFEX_USERNAME: sphinxjp
+        SPHINXINTL_TRANSIFEX_PROJECT_NAME: sphinx-doc
+      run: |
+        sh ./locale/update.sh
+    - name: git_config
+      run: |
+        git branch -a
+        git checkout master
```
Comment (tk0miya) [inline .github/workflows/main.yml:40]: I don't consider these two lines above is not a part of "git_config". How about move them into "commit" step?

File: .github/workflows/main.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+      uses: Tiryoh/gha-jobid-action@v0
+      id: jobs
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        job_name: script
+    - name: checkout with submodule
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+        token: ${{ secrets.PERSONAL_ACCESSTOKEN }}
```
Comment (tk0miya) [inline .github/workflows/main.yml:19]: Why this needed? It seems the default value of this action is `${{ github.token }}`. I feel no reason to change it to checkout this repository.
https://github.com/actions/checkout

File: .github/workflows/main.yml (hunk 9 above)
Comment (tkoyama010) [reply .github/workflows/main.yml:30]: I agree.
```suggestion
    - name: update
```

File: .github/workflows/main.yml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    - name: update
+      env:
+        TX_TOKEN: ${{ secrets.TX_TOKEN }}
+        SPHINXINTL_TRANSIFEX_USERNAME: sphinxjp
+        SPHINXINTL_TRANSIFEX_PROJECT_NAME: sphinx-doc
+      run: |
+        sh ./locale/update.sh
+    - name: git_config
+      run: |
+        git branch -a
```
Comment (tkoyama010) [inline .github/workflows/main.yml:39]: :wastebasket:

Comment (tkoyama010) [reply .github/workflows/main.yml:39]:
```suggestion
```

**Comment by rffontenelle** [comment]
Have you considered building translated docs after fetching the translation files from Transifex? This could be useful to find ReStructuredText syntax warnings or errors on translations.

I envisioned something parallelized, like one workflow per language workflow and with a 'push' event filtered by the proper path. For instance, to trigger build statements for pt_BR, the event would be:

```yaml
on:
  push:
    paths:
      - 'locale/pt_BR/LC_MESSAGES/*.po'
      - 'locale/pt_BR/LC_MESSAGES/**/*.po'
```

**Comment by shimizukawa** [comment]
@tkoyama010 and everybody, I am very sorry.
I created and merged PR #12 for GitHub Actions without noticing this PR.

**Comment by tkoyama010** [comment]
All good. Thank you for your work!

**Pull request closed without merging**